WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationFormViewModel.swift

```diff
@@ -109,6 +160,73 @@ final class BlazeCampaignCreationFormViewModel: ObservableObject {
     }
 }
 
+// MARK: Image download
+extension BlazeCampaignCreationFormViewModel {
+    func downloadProductImage() async {
+        image = await loadProductImage()
+    }
+}
+
+private extension BlazeCampaignCreationFormViewModel {
+    func loadProductImage() async -> MediaPickerImage? {
+        await withCheckedContinuation({ continuation in
+            guard let firstImage = product?.images.first else {
+                return continuation.resume(returning: nil)
+            }
+            _ = productImageLoader.requestImage(productImage: firstImage) { image in
+                continuation.resume(returning: .init(image: image, source: .productImage(image: firstImage)))
+            }
+        })
+    }
+}
+
+// MARK: - Blaze AI Suggestions
+extension BlazeCampaignCreationFormViewModel {
+    func loadAISuggestions() async {
+        isLoadingAISuggestions = true
+        error = nil
+
+        do {
+            suggestions = try await fetchAISuggestions()
+            if let firstSuggestion = suggestions.first {
+                tagline = firstSuggestion.siteName
+                description = firstSuggestion.textSnippet
+            }
+        } catch {
+            DDLogError("⛔️ Error fetching Blaze AI suggestions: \(error)")
+            self.error = .failedToLoadAISuggestions
```

Comment:
Super nit: the `self` is redundant right?

Reply:
`self` is used here as we also have a local `error` that we get from the catch block.